### PR TITLE
feat: add command-line interface

### PR DESCRIPTION
**Pull request by Breigner01**
## Description

Adds cli to goflowmeter

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Documentation update

## Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

## Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

I tested the different command line flags with both valid and invalid values.
Using strings for integer values throws an error and using values out of the range for ports throws also an error, as well as using an invalid log level (might need to modify the list of log levels in the future).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -19,6 +20,14 @@ func main() {
 	// Register signals to be notified
 	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
 
+	options, err := cli.ParseCLIFlags()
+	if err != nil {
+		panic(err)
```
Comment (0xpanoramix) [inline main.go:25]: log the error instead using the log package.

File: main.go [hunk 2]
```diff
@@ -19,6 +20,14 @@ func main() {
...
+	}
+
+	// To be removed
+	fmt.Printf("%+v\n", options)
```
Comment (0xpanoramix) [inline main.go:29]: Instead, use option.filename and use the pcap filename in the pcap.NewReader("HERE")

File: main.go (hunk 1 above)
Comment (Breigner01) [reply main.go]: Done

File: main.go (hunk 2 above)
Comment (Breigner01) [reply main.go]: Done

**Review by 0xpanoramix — APPROVED** [review]

**Pull request merged**